test/unit/objects/polling_test.py

```diff
@@ -272,3 +279,52 @@ def test_wait_for_event_finished_creation(
         assert len(get_requests) == 3
         assert result.entity.id == 11111
         assert result.status == "finished"
+
+    @httpretty.activate
+    def test_wait_for_event_finished_failed(
```

Comment:
I opted not to implement an integration test for this change because we generally shouldn't rely on runtime errors.